lib/chef/resource.rb

```diff
@@ -1627,6 +1635,15 @@ def lookup_provider_constant(name, action = :nothing)
     def suppress_up_to_date_messages?
       false
     end
+
+    private
+
+    def http_request_errors
+      source.map do |msg|
+        uri = URI.parse(msg)
+        "Error connecting to #{msg} - Failed to open TCP connection to #{uri.host}:#{uri.port}"
+      end
+    end
```

Comment:
all the code here in the resource class should go away.  i know the user request was to magically have the retries + retry_delay resource properties wired up to the Chef::HTTP object but that's the wrong abstraction and the wrong API.  those properties have a different meaning and the control of the HTTP settings needs to be done via a different API.

Reply:
Moved this code to remote_file resource.